pkg/auth/service.go

```diff
@@ -1933,7 +1962,78 @@ func (a *APIAuthService) CheckHealth(ctx context.Context, logger logging.Logger,
 	return nil
 }
 
-func NewAPIAuthService(apiEndpoint, token string, secretStore crypt.SecretStore, cacheConf params.ServiceCache, logger logging.Logger) (*APIAuthService, error) {
+func (a *APIAuthService) IsExternalPrincipalsEnabled(ctx context.Context) bool {
+	return a.externalPrincipalseEnabled
+}
+func (a *APIAuthService) CreateUserExternalPrincipal(ctx context.Context, userID, principalID string) error {
```

Comment:
I'm not sure if we need to check here user's existence.
WDYT?

Reply:
If you look at other requests such as `AttachPolicyToUser` and `DetachPolicyFromUser` when using `APIAuthService` (API service) we let that API decide on the correct authorization. 
Especially around external principal implementation i'd rather let the auth service dictate the business logic since the implementation details belong to there.